ds-caselaw-ingester/lambda_function.py

```diff
@@ -469,108 +378,294 @@ def process_message(message):
 
-    has_TDR_data = "TDR" in metadata["parameters"].keys()
-    if has_TDR_data:
-        store_metadata(uri, metadata)
-
-    # Determine if there's a word document -- we need to know before we save the tar.gz file
-    docx_filename = extract_docx_filename(metadata, consignment_reference)
-    print(f"extracted docx filename is {docx_filename!r}")
-
-    # Copy original tarfile
-    modified_targz_filename = (
-        filename if docx_filename else modify_filename(filename, "_nodocx")
-    )
-    store_file(
-        open(filename, mode="rb"),
-        uri,
-        os.path.basename(modified_targz_filename),
-        s3_client,
-    )
-    print(f"saved tar.gz as {modified_targz_filename!r}")
-
-    # Store docx and rename
-    # The docx_filename is None for files which have been reparsed.
-    if docx_filename is not None:
-        copy_file(
-            tar,
-            f"{consignment_reference}/{docx_filename}",
-            f'{uri.replace("/", "_")}.docx',
-            uri,
-            s3_client,
+    def update_document_xml(self) -> bool:
+        if self.metadata_object.is_tdr:
+            message = "Updated document submitted by TDR user"
+        else:
+            message = "Updated document uploaded by Find Case Law"
+        try:
+            annotation = VersionAnnotation(
+                VersionType.SUBMISSION,
+                automated=self.metadata_object.force_publish,
+                message=message,
+                payload=_build_version_annotation_payload_from_metadata(self.metadata),
+            )
+
+            api_client.get_judgment_xml(self.uri, show_unpublished=True)
+            api_client.update_document_xml(self.uri, self.xml, annotation)
+            return True
+        except MarklogicResourceNotFoundError:
+            return False
+
+    def insert_document_xml(self) -> bool:
+        if self.metadata_object.is_tdr:
+            message = "New document submitted by TDR user"
+        else:
+            message = "New document uploaded by Find Case Law"
+        annotation = VersionAnnotation(
+            VersionType.SUBMISSION,
+            automated=self.metadata_object.force_publish,
+            message=message,
+            payload=_build_version_annotation_payload_from_metadata(self.metadata),
         )
+        api_client.insert_document_xml(self.uri, self.xml, annotation)
+        return True
 
-    # Store parser log
-    try:
-        copy_file(
-            tar, f"{consignment_reference}/parser.log", "parser.log", uri, s3_client
+    def send_updated_judgment_notification(self) -> None:
+        personalisation = personalise_email(self.uri, self.metadata)
+        if os.getenv("ROLLBAR_ENV") != "prod":
+            print(
+                f"Would send a notification but we're not in production.\n{personalisation}"
+            )
+            return
+
+        notifications_client = NotificationsAPIClient(os.getenv("NOTIFY_API_KEY"))
```

Comment:
What error is thrown if API keys are not configured? Not a huge problem, but just might make it quicker to debug if there is some level of validation that throws configuration errors detailing the missing variable.

Reply:
`os.getenv` returns no error; NotificationsAPIClient returns a super cryptic error message (`TypeError: 'NoneType' object is not subscriptable`)

`os.environ['NOTIFY_API_KEY']` would return `KeyError: 'NOTIFY_API_KEY'` which is slightly less cryptic.

Now realising why Enrichment has such overwrought environment management functions :)

Reply:
I would be tempted to go for the latter `os.environ['NOTIFY_API_KEY']` since the key error makes it pretty easily identifiable in the logs.